### PR TITLE
backend: add lightning balance to total balance

### PR DESCRIPTION
**Pull request by strmci**
This commit fixes calculation of totals in my portfolio when the lightning wallet is active.

### CODE REVIEW

File: backend/backend.go [hunk 1]
```diff
@@ -642,6 +642,17 @@ func (backend *Backend) AccountsTotalBalanceByKeystore() (map[string]KeystoreTot
 			)
 			currentTotal.Add(currentTotal, fiatValue)
 		}
+		// Add lightning balance if it is enabled.
+		if len(backend.Config().LightningConfig().Accounts) > 0 {
```
Comment (Beerosagos) [inline backend/backend.go:646]: Could you please add a `lightningConfig.LightningEnabled()` method that returns `len(backend.Config().LightningConfig().Accounts) > 0` inside config/lightning.go, and use it to check if lightning is enabled? this way we avoid duplicating the check around the code. We may want to update it in the future.

File: backend/backend.go [hunk 2]
```diff
@@ -650,6 +661,33 @@ func (backend *Backend) AccountsTotalBalanceByKeystore() (map[string]KeystoreTot
 	return totalAmounts, nil
 }
 
+// updateTotalWithLightningBalance adds lightning balance to current fiat total.
+func (backend *Backend) updateTotalWithLightningBalance(currentTotal *big.Rat) (*big.Rat, error) {
```
[the remark this answers is not in the document]
Comment (Beerosagos) [reply backend/backend.go:665]: Also, it would be nice to have a unit test for this! :innocent:

File: backend/backend.go [hunk 3]
```diff
@@ -650,6 +661,33 @@ func (backend *Backend) AccountsTotalBalanceByKeystore() (map[string]KeystoreTot
...
+	fiat := backend.Config().AppConfig().Backend.MainFiat
+	coinDecimals := new(big.Int).Exp(
+		big.NewInt(10),
+		big.NewInt(int64(8)),
```
Comment (Beerosagos) [inline backend/backend.go:669]: You can retrieve the BTC coin with `backend.Coin(coinpkg.CodeBTC)` and use it to retrieve the values for decimals and "BTC" unit.

**Review by Beerosagos — CHANGES_REQUESTED** [review]

File: backend/backend.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (strmci) [reply backend/backend.go]: Created new function `convertAmountToFiat` with a unit test that is used in `AccountsTotalBalanceByKeystore` and `ChartData`

File: backend/backend.go (hunk 1 above)
Comment (strmci) [reply backend/backend.go]: that will be better, `LightningEnabled` added

File: backend/backend.go (hunk 3 above)
Comment (strmci) [reply backend/backend.go]: I added the `backend.Coin(coinpkg.CodeBTC)` to the `convertAmountToFiat` function, now the function is only for BTC amounts, shall we make it work for other coins?

Comment (Beerosagos) [reply backend/backend.go]: We could make it work with any coin, or rename it to `convertBtcAmountToFiat`. Maybe keep it only on BTC is good for now, we can always extend. Also, generic backend refactoring are not in the scope of the branch, and makes merging the master more difficult

File: backend/accounts_test.go [hunk 4]
```diff
@@ -1506,3 +1507,17 @@ func TestAccountsTotalBalanceByKeystore(t *testing.T) {
...
+	b := newBackend(t, testnetDisabled, regtestDisabled)
+	defer b.Close()
+	b.ratesUpdater = rates.MockRateUpdater()
+	defer b.ratesUpdater.Stop()
+	// convert 1 BTC to USD
+	amount := coinpkg.NewAmountFromInt64(100000000)
+	converted, err := b.convertAmountToFiat(amount, "USD")
+	expectedValue := big.NewRat(21, 1)
+
+	require.NoError(t, err)
```
Comment (Beerosagos) [inline backend/accounts_test.go:1521]: nit: this line should be right below b.convertAmountToFiat for better readability

File: backend/accounts_test.go [hunk 5]
```diff
@@ -1506,3 +1507,17 @@ func TestAccountsTotalBalanceByKeystore(t *testing.T) {
...
 }
+
+func TestConvertAmountToFiat(t *testing.T) {
+	b := newBackend(t, testnetDisabled, regtestDisabled)
+	defer b.Close()
+	b.ratesUpdater = rates.MockRateUpdater()
+	defer b.ratesUpdater.Stop()
+	// convert 1 BTC to USD
+	amount := coinpkg.NewAmountFromInt64(100000000)
+	converted, err := b.convertAmountToFiat(amount, "USD")
```
Comment (Beerosagos) [inline backend/accounts_test.go:1518]: nit: "USD" -> rates.USD.String()

**Review by Beerosagos — CHANGES_REQUESTED** [review]

**Comment by strmci** [comment]
@Beerosagos I renamed the function to `convertBtcAmountToFiat` and made all the suggested changes, PTAL

**Pull request merged**